### DISCUSSION TITLE
Likes system

### DISCUSSION THREAD

**Discussion by DZ8540** [adonisjs/core#3675] [Help]
I need help with make like system in adonisJS application.

I tried make likeStatus in model

<img width="365" alt="image" src="https://user-images.githubusercontent.com/50152205/163214697-495d0371-cfe2-4625-bf5b-0b68d90b98ce.png">

And make method for use likeStatus

<img width="884" alt="image" src="https://user-images.githubusercontent.com/50152205/163214832-843a6c99-c1ff-40d8-8af7-9e344acf1860.png">

And use it like this

<img width="412" alt="image" src="https://user-images.githubusercontent.com/50152205/163214929-7248f867-d19c-4c9a-9e6d-74884df1fae0.png">

And... this is works! But! When i'm used CRUD operations except get from DB, application throwed error like this

<img width="517" alt="image" src="https://user-images.githubusercontent.com/50152205/163215381-ed413307-83e1-4340-908c-8f6c4f98f2af.png">

I suggested to my employee to keep all likes array with other used data, when it logged in, and set like for posts and stories in frontend side. But he told me it's not right because this array might be too big. 

What is the right thing to do in this situation?

**Comment by galaczi** [comment]
If I understand you correctly, you want to check if a user liked a post. In that case why not simply use many to many relationship and query that user - item combination? If it exists, the user liked the post.

**Reply by DZ8540** [reply]
Yes, but i send all posts via pagination to main page, and i dont know how to set like status to every post

**Reply by galaczi** [reply]
You can preload the likes on the posts and then paginate them. Something like:

```
const posts = await Post
  .query()
  .preload('likes')
  .paginate(page, limit)
```

**Reply by DZ8540** [reply]
This is preloaded all likes from all users, is there an other way?

**Reply by galaczi** [reply]
I am pretty sure it preloads only the likes belonging to that page.

**Reply by DZ8540** [reply]
For each post on the current page, all users who like each post on the current page will be preloaded. The request will be very heavy, i can of course selected only users' id inside preload query, but query still be very heavy for this way(

**Reply by DZ8540** [reply]
It would be great if i could make a request to search the table for the id of the current user, and then just attach the status to the post, but i dont know how
